periodic/periodic.go

```diff
@@ -497,9 +502,19 @@ func runOne(id int, runnerChan chan struct{},
 	tIDStr := fmt.Sprintf("T%03d", id)
 	perThreadQPS := r.QPS / float64(r.NumThreads)
 	useQPS := (perThreadQPS > 0)
+
 	hasDuration := (r.Duration > 0)
 	useExactly := (r.Exactly > 0)
 	f := r.Runners[id]
+	if useQPS && r.Uniform {
+		delayBetweenRequest := 1. / float64(perThreadQPS)
+		// When using uniform mode, we should wait a bit relative to our QPS and thread ID.
+		// For example, with 10 threads and 1 QPS, thread 8 should delay 0.7s.
+		delaySeconds := delayBetweenRequest - (delayBetweenRequest / float64(r.NumThreads) * float64(r.NumThreads-id))
+		delayDuration := time.Duration(delaySeconds * float64(time.Second))
```

Comment:
Do you think this still works:
https://github.com/fortio/fortio/blob/v1.19.0/periodic/periodic.go#L529-L532
can you try/does it invalidate it (and maybe it's fine if it does for -uniform and low duration corner cases but I am curious)

Reply:
I think with uniform we need to compromise on QPS or duration - we cannot respect both. Say we have 2 threads, 1 qps, 2s duration

Today we do:
```
         0s   01   2s
thread0: .         .
thread1: .         .
```

This PR does this - favoring exact QPS over exact duration
```
         0s   01   2s    3s
thread0: .         .
thread1:       .         .
```
So it goes past the duration by `delayDuration`.

The alternative would be to slightly increase the QPS to hit the target duration:
```
         0s   01   2s 
thread0: .     .
thread1:    .       .
```

But that seems less desired

Reply:
agreed, thx!